### ISSUE TITLE
Connect-AzContainerRegistry issue: given key not present in dictionary

### ISSUE THREAD

**Issue by Willem-J-an** [Azure/azure-powershell#20742]
### Description

Issue when connecting container registry in 3.0.1

Connect-AzContainerRegistry -Name $AzureContainerRegistryName
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The given key 'ContainerRegistryEndpointResourceId' was not present in the dictionary.

Possibly related to #20528?

### Issue script & Debug output

```PowerShell
Connect-AzContainerRegistry -Name $AzureContainerRegistryName -Debug:$true
3:39:11 PM - ConnectAzureContainerRegistry begin processing with ParameterSet 'WithoutNameAndPasswordParameterSet'.
3:39:11 PM - using account id 'stuff'...
3:39:11 PM - [ConfigManager] Got nothing from [DisplayBreakingChangeWarning], Module = [], Cmdlet = []. Returning default value [True].
3:39:11 PM - [ConfigManager] Got nothing from [EnableDataCollection], Module = [], Cmdlet = []. Returning default value [True].

Connect-AzContainerRegistry: The given key 'ContainerRegistryEndpointResourceId' was not present in the dictionary.
3:39:11 PM - [ConfigManager] Got nothing from [DisplayBreakingChangeWarning], Module = [], Cmdlet = []. Returning default value [True].
AzureQoSEvent:  Module: Az.ContainerRegistry:3.0.1; CommandName: Connect-AzContainerRegistry; PSVersion: 7.3.1; IsSuccess: False; Duration: 00:00:00.0143032; Exception: The given key 'ContainerRegistryEndpointResourceId' was not present in the dictionary.;
3:39:11 PM - ConnectAzureContainerRegistry end processing.
```

### Environment data

```PowerShell
Name                           Value
----                           -----
PSVersion                      7.3.1
PSEdition                      Core
GitCommitId                    7.3.1
OS                             Linux 5.10.0-20-amd64 #1 SMP Debian 5.10.158-2 (2022-12-13)
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Module versions

```PowerShell
Script     2.11.1                Az.Accounts
Script     3.0.1                 Az.ContainerRegistry
```

### Error output

```PowerShell
HistoryId: 8

Message        : The given key 'ContainerRegistryEndpointResourceId' was not present in the dictionary.
StackTrace     :    at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
                    at Microsoft.Azure.Commands.ContainerRegistry.ContainerRegistryDataPlaneClient.SetEndPoint(String RegistryName)
                    at Microsoft.Azure.Commands.ContainerRegistry.ConnectAzureContainerRegistry.ExecuteCmdlet()
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.<>c__3`1.<ExecuteSynchronouslyOrAsJob>b__3_0(T c)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet, Action`1 executor)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
Exception      : System.Collections.Generic.KeyNotFoundException
InvocationInfo : {Connect-AzContainerRegistry}
Line           : Connect-AzContainerRegistry -Name $AzureContainerRegistryName -Debug:$true
Position       : At line:1 char:1
                 + Connect-AzContainerRegistry -Name $AzureContainerRegistryName -Debug: …
                 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 8


   HistoryId: 4

Message        : The given key 'ContainerRegistryEndpointResourceId' was not present in the dictionary.
StackTrace     :    at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
                    at Microsoft.Azure.Commands.ContainerRegistry.ContainerRegistryDataPlaneClient.SetEndPoint(String RegistryName)
                    at Microsoft.Azure.Commands.ContainerRegistry.ConnectAzureContainerRegistry.ExecuteCmdlet()
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.<>c__3`1.<ExecuteSynchronouslyOrAsJob>b__3_0(T c)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet, Action`1 executor)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
Exception      : System.Collections.Generic.KeyNotFoundException
InvocationInfo : {Connect-AzContainerRegistry}
Line           : Connect-AzContainerRegistry -Name $AzureContainerRegistryName
Position       : At line:1 char:1
                 + Connect-AzContainerRegistry -Name $AzureContainerRegistryName
                 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 4


   HistoryId: 3
```

**Comment by RakeshMohanMSFT** [comment]
@Willem-J-an Thank you for reaching out, we will look into it.

**Comment by RakeshMohanMSFT** [comment]
@Willem-J-an We are unable to reproduce this issue at our end. Can you please try to reinstall the container registry module? and retry operation
![image](https://user-images.githubusercontent.com/49954584/215258995-3dddeeec-18f9-4dde-b676-45a360a817a9.png)

**Comment by Willem-J-an** [comment]
I removed the module, installed it again; the same issue is still happening consistently. The problem is not there on version 3.0.0, but version 3.0.1 introduces this issue.

**Comment by pownkel** [comment]
I'm running into the same issue, have there been any updates?

**Comment by msuleman92** [comment]
any update on this?

**Comment by Willem-J-an** [comment]
It seems fixed for me on 3.0.3; although I also migrated to another distribution, so your mileage may vary.
![image](https://user-images.githubusercontent.com/51120533/235146601-234d3893-5486-4e16-8d78-82314beaf179.png)

Closing this.
